### PR TITLE
Created FAQ.md fixes #162 and fixes #172

### LINKED ISSUE #162: Provide Gradle 2.4 instructions: Eclipse (Gradle 2.2.1) does not work with j2objc-gradle plugin
I created a new fresh sample project in eclipse with the eclipse gradle plugin. Then I imported: 

```
apply plugin: 'com.github.j2objccontrib.j2objcgradle'
buildscript {
    classpath files('/Users/mg/Downloads/j2objc-gradle/build/libs/j2objc-gradle-0.2.2-alpha.jar')
  }
}
```
After refresh Eclipse raises the following error: 

problem_occurred

and after creating ``local.properties``: 

multiple_problems_have_occurred

Console output from Eclipse is:

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'test'.
> Exception thrown while executing model rule: model.platforms
   > Cannot convert the provided notation to an object of type Architecture: ios_arm64.
     The following types/formats are supported:
       - Instances of ArchitectureInternal.
       - One of the following values: 'x86', 'i386', 'ia-32', 'x86_64', 'amd64', 'x64', 'x86-64', 'ia-64', 'ppc', 'ppc64', 'sparc', 'sparc32', 'sparc-v7', 'sparc-v8', 'sparc64', 'ultrasparc', 'sparc-v9', 'arm'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 0.223 secs
```

### LINKED ISSUE #172: Files from generatedSourceDirs are not included correctly
I use two projects: ``base `` and ``main``. Where main includes base. 

Problem: I copied the sources of base into main "build/source/base". Main uses the annotation processor of base and generates "build/source/apt". It turns out that files in these two folders are only included when the are used in main project's ``src/main/java``. I need all the files in these folders to be translated. Using the setting ``generatedSourceDirs`` should do that. 

Here is my configuration:

settings.gradle in main: 
```
include ':base'
project(':base').projectDir = new File('../base')
```

build.gradle in main: 
```
apply plugin: 'java'
apply plugin: 'apt'
apply plugin: 'fr.putnami.gwt-lib'
apply plugin: 'eclipse'
apply plugin: 'com.github.j2objccontrib.j2objcgradle'


group = 'core'
sourceCompatibility = 1.7
version = '1.0'

eclipse {
  classpath {
     downloadSources=true
     downloadJavadoc=true
  }
}

buildscript {
  repositories {
    mavenCentral()
    jcenter()
    maven {
      url "https://oss.sonatype.org/content/repositories/snapshots/"
    }
    maven {
      url uri('../../0Gradle-Repo')
    }
  }
  dependencies {
    classpath 'com.jimdo.gradle:gradle-apt-plugin:0.5-SNAPSHOT'  
    classpath files('/Users/mg/Downloads/j2objc-gradle/build/libs/j2objc-gradle-0.2.2-alpha.jar')
  }
}

 
repositories {
  mavenCentral()
  maven {
    name = "sonatype"
    url = "https://oss.sonatype.org/content/repositories/snapshots/"
  }
  maven {
    name = "sonatypeGoogle"
    url = "https://oss.sonatype.org/content/repositories/google-snapshots/"
  }
  flatDir {
    dirs 'lib'
  }
}


sourceSets {
  apt{
    java{
      srcDir 'build/source/apt'
      srcDir 'build/source/base'
    }
  }
}

dependencies {
  apt (project(':base'))
  apt 'com.google.auto.factory:auto-factory:1.0-beta2'
  apt 'com.google.dagger:dagger:2.0.1'
  apt 'com.google.dagger:dagger-producers:2.0-beta'
  apt 'com.google.dagger:dagger-compiler:2.0.1'

  compile (project('base'))
  compile 'com.google.guava:guava:14.0.1' 
  compile 'javax.inject:javax.inject:1' 
  compile 'org.json:json:20140107'
  
  compile 'com.google.auto.factory:auto-factory:1.0-beta2'
  compile 'com.google.auto.service:auto-service:1.0-rc2'
  
  compile name: 'j2objc_annotations'
  compile 'com.google.dagger:dagger:2.0.1'
}

jar {
  from "build/source/apt"
  exclude '**/*.zip'
}


// custom tasks

task copyBaseSources(type: Copy) {
  description "Copy all source files from base into the current project."
  exclude '**/*processor*'  // Annotation Processors should not be copied and translated
  from project(":base").projectDir.toString() + "/src/main/java"
  into project.projectDir.toString() + "/build/source/base"
}

tasks.compileJava.dependsOn += copyBaseSources


// ------------


j2objcConfig {
  generatedSourceDirs "build/source/apt", "build/source/base"
  
  xcodeProjectDir "${projectDir}/../Buddyis"
  xcodeTarget "Buddyis"

  
 
  translateFlags "--no-package-directories --prefixes prefixes.properties -use-arc --doc-comments --build-closure --generate-deprecated"
  
  translateClassPaths "json-20140107.jar", "javax.inject-1.jar", "auto-factory-1.0-beta2.jar", "auto-service-1.0-rc2.jar", "javawriter-2.5.1.jar"
  translateSourcepaths "${projectDir}/libSrc/dagger-2.0.1-sources.jar"
  
  translateJ2objcLibs "j2objc_annotations.jar", "j2objc_guava.jar", "javax.inject-1.jar", "jsr305-3.0.0.jar"
   
  
  filenameCollisionCheck = true
  cycleFinderExpectedCycles 0
  
  cycleFinderFlags "--whitelist /Applications/J2Objc/j2objc-sorce/jre_emul/cycle_whitelist.txt --sourcefilelist /Applications/J2Objc/j2objc-sorce/jre_emul/build_result/java_sources.mf"
  
} 
    
```

### PR DESCRIPTION
**Pull request by confile**

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# FAQ
+
+### Eclipse raises error ``Obtaining Gradle model...``?
+
+You have to create the gralde wrapper first. Go to your project folder and do ``gradle wrapper``. Refresh your eclipse project.
```
Comment (brunobowden) [inline FAQ.md]: gralde => gradle

**Comment by brunobowden** [comment]
LGTM - make that fix and I'll merge in

File: FAQ.md (hunk 1 above)
Comment (confile) [inline FAQ.md]: fixed

**Comment by confile** [comment]
Wait I just squash the commits.

**Comment by brunobowden** [comment]
It would be better if we could automatically detect problems and warn that way... so someone doesn't have to read the FAQ... but fine for now.

**Comment by confile** [comment]
@brunobowden @advayDev1  Thanks for all your help. It is a lot of fun working with you I learn a lot. Don't hesitate criticizing me.

**Pull request merged**